app/services/authorizer.rb

```diff
@@ -12,8 +12,9 @@ def can?(permission, subject = nil)
       user.permissions.where(:name => permission).present?
     else
       return true if user.admin?
-      collection = @cache[subject.class.to_s][permission] ||= find_collection(subject.class, :permission => permission)
-      collection.include?(subject)
+      @cache["#{subject.class.to_s}-#{permission}-#{subject.id}"] ||=
+        find_collection(subject.class, :permission => permission).
+        where(:id => subject.id)
```

Comment:
Yes, I do partly agree that the collection may be obsolete or could be optimised further, though removing it now will require more care to avoid continually reloading filters, permissions etc on index pages.  I'd prefer to look at that separately as it's optimising for the opposite (and more common) case to this.